### PR TITLE
[IOPAE-1398] Fetch vehicles

### PR DESCRIPTION
**Pull request by adelloste**
This PR implements the logic to retrieve the list of vehicles by fiscal code.

#### List of Changes

- Added `template-bff.hbs` template to generate types from the schema
- Added `@tanstack/react-query`
- Added `@tanstack/eslint-plugin-query`
- Added testing network call

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/my-vehicles/src/hooks/useVehicles.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      codiceFiscaleProprietario: fiscalCode,
+    },
+  });
+  return response;
+};
+
+export const useVehicles = (fiscalCode: string) =>
+  useQuery({
+    queryFn: () => fetchVehicles(fiscalCode),
+    queryKey: ["vehicles"],
```
Comment (gunzip) [inline apps/my-vehicles/src/hooks/useVehicles.ts:16]: queryKey Should Include all dependencies, ie. `queryKey: ["vehicles", fiscalCode]`

**Review by gunzip — COMMENTED** [review]
lacks unit tests

**Review by giuseppedipinto — CHANGES_REQUESTED** [review]

File: apps/my-vehicles/src/hooks/useVehicles.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { client } from "@/common/client";
+import { useQuery } from "@tanstack/react-query";
+
+const fetchVehicles = async (fiscalCode: string) => {
```
[the remark this answers is not in the document]
Comment (adelloste) [reply apps/my-vehicles/src/hooks/useVehicles.ts]: Removed `fiscalCode` from bff openapi and updated `fetchVehicles`

File: apps/my-vehicles/package.json [hunk 3]
```diff
@@ -17,9 +17,11 @@
     "build": "next build",
     "start": "next start",
     "generate": "npm-run-all 'generate:*'",
-    "generate:openapi-schema": "rimraf src/generated && mkdir -p src/generated && openapi-zod-client ./openapi.yaml -o src/generated/openapi.ts --export-schemas --additional-props-default-value=false"
+    "generate:openapi-schema": "rimraf src/generated && mkdir -p src/generated && openapi-zod-client ./openapi.yaml -o src/generated/openapi.ts -t template-bff.hbs --additional-props-default-value=false"
   },
   "dependencies": {
+    "@tanstack/react-query": "^5.56.2",
+    "@tanstack/react-query-devtools": "^5.56.2",
```
[the remark this answers is not in the document]
Comment (adelloste) [reply apps/my-vehicles/package.json:24]: according to the doc, we can add @tanstack/react-query-devtools as a dependency

**Review by giuseppedipinto — APPROVED** [review]
LGTM

**Pull request merged**